Mage.Sets/src/mage/cards/s/SilverScrutiny.java

```diff
@@ -0,0 +1,154 @@
+package mage.cards.s;
+
+import mage.MageObject;
+import mage.abilities.SpellAbility;
+import mage.abilities.costs.mana.VariableManaCost;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.InfoEffect;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.AdventureCardSpell;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AsThoughEffectType;
+import mage.constants.CardType;
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.game.stack.SpellStack;
+import mage.game.stack.StackObject;
+import mage.watchers.Watcher;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class SilverScrutiny extends CardImpl {
+
+    public SilverScrutiny(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{U}{U}");
+
+        // You may cast Silver Scrutiny as though it had flash if X is 3 or less.
+        // Draw X cards.
+        this.replaceSpellAbility(new SilverScrutinySpellAbility(this));
+    }
+
+    private SilverScrutiny(final SilverScrutiny card) {
+        super(card);
+    }
+
+    @Override
+    public SilverScrutiny copy() {
+        return new SilverScrutiny(this);
+    }
+}
+
+class SilverScrutinySpellAbility extends SpellAbility {
+
+    public SilverScrutinySpellAbility(Card card) {
+        super(card.getManaCost(), card.getName());
+        this.addEffect(new InfoEffect("You mast cast {this} as though it had flash if X is 3 or less.<br>"));
+        this.addEffect(new DrawCardSourceControllerEffect(ManacostVariableValue.REGULAR));
+        this.addWatcher(new SilverScrutinyWatcher());
+    }
+
+    private SilverScrutinySpellAbility(final SilverScrutinySpellAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public SilverScrutinySpellAbility copy() {
+        return new SilverScrutinySpellAbility(this);
+    }
+
+    @Override
+    public boolean activate(Game game, boolean noMana) {
+        SilverScrutinyWatcher watcher = game.getState().getWatcher(SilverScrutinyWatcher.class);
+        if (watcher != null && watcher.spellCastAsInstant(sourceId)) {
+            for (Object cost : manaCostsToPay) {
+                if (cost instanceof VariableManaCost) {
+                    ((VariableManaCost) cost).setMaxX(3);
+                }
+            }
+        }
+        return super.activate(game, noMana);
+    }
+}
+
+class SilverScrutinyWatcher extends Watcher {
```

Comment:
I think this class needs some flattening, and a documentation of what it's doing. I currently cannot understand it.

Reply:
It keeps a HashSet of spells that were cast that could not normally be cast.  It's basically the same code as in `SpellAbility.spellCanBeActivatedRegularlyNow` with a couple modifications to make it work as a Watcher.

The `PlayFromNotOwnHandZone` check if for cards like Chandra that allow you to cast it not as a sorcery (in which case, the max X value would not apply).

If it sees the player cast a spell at instant speed without any other effect allowing that to happen, we can assume they're using Silver Scrutiny's ability to cast with flash.  The spell gets then gets added to the HashMap which is read by the SpellAbility's activate method.